### ISSUE TITLE
Sprite Area

### ISSUE THREAD

**Issue by SillyInventor** [LLK/scratch-gui#53]
### Expected Behavior

Should be able to edit sprite name, draggable, location, visbility, and rotation style (manually)

### Actual Behavior

Functionality not added

### Operating System and Browser

Ubuntu 16.04, chrome 56.0.2924.76 

I am sure you're already working on this, just adding an issue so I can share our first play test findings / complaints.

**Comment by carljbowman** [comment]
Assigning to myself to drop in mocks and design specs

**Comment by carljbowman** [comment]
# 2.0 - Sprite Area
<img width="412" alt="screen shot 2017-02-07 at 9 46 35 am" src="https://cloud.githubusercontent.com/assets/3409578/22695891/58f19ed6-ed1a-11e6-8c8c-cde2a088e5b4.png">

## Add Sprite 
<img width="183" alt="screen shot 2017-02-07 at 9 38 50 am" src="https://cloud.githubusercontent.com/assets/3409578/22695582/3f8079b4-ed19-11e6-8a01-d50a8c795e61.png">

- via Library
- via Creating in Paint Editor
- via Local File Upload
- via Photo Capture

## Selection
![sprite selection](https://cloud.githubusercontent.com/assets/3409578/22696097/e33e68ee-ed1a-11e6-9aca-5a4b85079360.gif)
- Selected
- Unselected

## Drag & Drop Reordering
![sprite reorder](https://cloud.githubusercontent.com/assets/3409578/22696445/05327ac0-ed1c-11e6-9362-0ad38d817ffc.gif)

## Context Menu (right-click or shift-click)
<img width="412" alt="screen shot 2017-02-07 at 10 01 36 am" src="https://cloud.githubusercontent.com/assets/3409578/22696606/9ee4dd98-ed1c-11e6-9614-11a158b243b0.png">

- Expose Sprite Info
- Duplicate Sprite
- Delete Sprite
- Save to local file
- Show/Hide toggle

## Exposing Sprite Info
![sprite info](https://cloud.githubusercontent.com/assets/3409578/22696370/bcd61264-ed1b-11e6-9c56-3ea739b5ac9f.gif)
- Info Icon
- Context menu option (right-click or shift-click)

## Sprite Info
<img width="417" alt="screen shot 2017-02-07 at 10 07 43 am" src="https://cloud.githubusercontent.com/assets/3409578/22696764/4950e326-ed1d-11e6-839b-03363e904d3d.png">

- Edit Sprite Name
- X, Y Position
- Direction UI
- Rotation Style Toggle
- Draggle Toggle
- Show/Hide Toggle

**Comment by carljbowman** [comment]
# 3.0 - Animation Playtest Prototype

### Overview
<img width="688" alt="screen shot 2017-02-07 at 9 34 40 am" src="https://cloud.githubusercontent.com/assets/3409578/22697119/7c007272-ed1e-11e6-9284-e75402bdd211.png">

#### In Scope
- [ ] Add Sprite
  - [ ] via Library
- [ ] Selection
  - [ ] Selected state
  - [ ] Unselected state
- [ ] Exposed Sprite Info
  - [ ] Edit Sprite name
  - [ ] Editable X, Y Position
  - [ ] Rotation Style Dropdown
  - [ ] Draggable Toggle
  - [ ] Show/Hide Toggle

#### Out of Scope
- Add Sprite 
  - via Creating in Paint Editor
  - via Local File Upload
  - via Photo Capture
- Context Menu
- Drag & Drop Reordering

---

![r1_gui_animate-prototype](https://cloud.githubusercontent.com/assets/3409578/22697557/c8047122-ed1f-11e6-8215-b79034bb80c5.png)

## Add Sprite via Library
<img width="408" alt="screen shot 2017-02-07 at 10 38 57 am" src="https://cloud.githubusercontent.com/assets/3409578/22698116/aae0efe2-ed21-11e6-95b2-110aa9ecfebc.png">

- Single button (bottom left) opens Sprite Library upon click
- Appears above (z-index) Sprite 'tiles'
- Could grow on hover, like 'Scratch' logo in the WWW Navigation

## Selection
<img width="410" alt="screen shot 2017-02-07 at 10 48 55 am" src="https://cloud.githubusercontent.com/assets/3409578/22698632/10eabb3c-ed23-11e6-93ed-59ef68d58bce.png">

- Tile UI similar to 2.0 styling (The 'i' Info icon is not so important for this prototype, but we explore that toggle the amount of Sprite Info displayed above)
- In addition to switching the workspace to the selected Sprite, the Sprite Info displayed above switches.
- Interactive states (e.g. Hover) TBD

## Exposed Sprite Info
The largest difference for this prototype is exploring exposing the Sprite Info. This area should update relative to the Sprite the sprite that is selected.

<img width="409" alt="screen shot 2017-02-07 at 11 10 02 am" src="https://cloud.githubusercontent.com/assets/3409578/22699629/ff697184-ed25-11e6-9966-13fc06f62109.png">

- The info should update based on User Interaction (e.g. dragging a sprite changing x,y) and update based on scripts (e.g. set rotation style)
- These fields can be altered directly (e.g. a user can type in a new x position and the sprite move on the stage)
- Field shape echo the Block Input shapes, we could also lean on the block interaction states (e.g Focus)

**Comment by liam4** [comment]
I wonder if it might be helpful to have a "drop shadow" below a dragged sprite (icon), similar to the way blocks work in Scratch 3.0?

![blocks](https://cloud.githubusercontent.com/assets/9948030/22704072/ce916816-ed3c-11e6-87d7-405bd4e15c15.gif)

Currently it's a little bit tough to figure out where a sprite (icon) will be dropped:

![sprites](https://cloud.githubusercontent.com/assets/9948030/22704094/dd11d2cc-ed3c-11e6-88a4-6fb210e0b77f.gif)

..especially if you're dragging things around quickly.

**Comment by Airhogs777** [comment]
either that or a "cursor" line between the two sprites that it would go between

**Comment by carljbowman** [comment]
@liam4 - Totally agree. We won't have Sprite 'drag & drop' within this initial pass on the Sprite Area, but very excited to iterate on on this interaction.

**Comment by lifeinchords** [comment]
TODO: Refactor flex layout of both rows, to better structure

**Comment by carljbowman** [comment]
Closing issue. Broke out remaining features into their own issues.

**Comment by liam4** [comment]
Not sure where to suggest this (I don't feel it deserves an issue unless it were to be considered worth taking a good look at by you guys), but the old "surprise sprite" button was pretty nice – here's a simple scene I made starting by clicking on the button a few times, and then continuing from there:

![Scene](https://cloud.githubusercontent.com/assets/9948030/23728307/8473bbaa-0432-11e7-8d44-c6ba3eac8a56.png)

I don't think it'd be too hard to implement, either!

**Comment by carljbowman** [comment]
@liam4 - Thanks for suggesting! We have talked about bringing this feature back. I certainly think it is worth exploring. 

On a similar note, I just had a chat with @ntlrsk today about exploring the a surprise/random feature at the block menu level too. For example in blocks like 'set costume to ( costume 1 v )', there would be an 'any' or 'surprise' option. 

We explored this menu option in the horizontal grammar with setting the LEGO WeDo LED color. There was a surprise option.
